pydrive2/auth.py

```diff
@@ -173,40 +77,100 @@ class GoogleAuth(ApiAttributeMixin):
     def __init__(self, settings_file="settings.yaml", http_timeout=None):
         """Create an instance of GoogleAuth.
 
-        This constructor just sets the path of settings file.
-        It does not actually read the file.
+        This constructor parses just he yaml settings file.
+        All other settings are lazy
 
         :param settings_file: path of settings file. 'settings.yaml' by default.
         :type settings_file: str.
         """
         self.http_timeout = http_timeout
         ApiAttributeMixin.__init__(self)
-        self.thread_local = threading.local()
-        self.client_config = {}
+
         try:
             self.settings = LoadSettingsFile(settings_file)
         except SettingsError:
-            self.settings = self.DEFAULT_SETTINGS
+            self.settings = DEFAULT_SETTINGS
         else:
-            if self.settings is None:
-                self.settings = self.DEFAULT_SETTINGS
+            # if no exceptions
+            ValidateSettings(self.settings)
+
+        self._storage = None
+        self._service = None
+        self._credentials = None
+        self._client_config = None
+        self._oauth_type = None
+        self._flow = None
+
+    # Lazy loading, read-only properties
+    @property
+    def service(self):
+        if not self._service:
+            self._service = build("drive", "v2", cache_discovery=False)
```

Comment:
Before it was providing auth-ed HTTP into this call? does it mean that it's not required?

Reply:
I believe this is not required as long as a http object is passed to the `execute` method of the service. The second example [here](https://github.com/googleapis/google-api-python-client/blob/main/docs/thread_safety.md) also created a service without a http